packages/agreement-process/src/services/agreementStateProcessor.ts

```diff
@@ -117,27 +117,11 @@ export const nextState = (
 export const agreementStateByFlags = (
   stateByAttribute: AgreementState,
   suspendedByProducer: boolean | undefined,
-  suspendedByConsumer: boolean | undefined,
-  suspendedByPlatform: boolean | undefined
+  suspendedByConsumer: boolean | undefined
 ): AgreementState =>
-  match([
-    stateByAttribute,
-    suspendedByProducer,
-    suspendedByConsumer,
-    suspendedByPlatform,
-  ])
-    .with(
-      [agreementState.active, true, P.any, P.any],
-      () => agreementState.suspended
-    )
-    .with(
-      [agreementState.active, P.any, true, P.any],
-      () => agreementState.suspended
-    )
-    .with(
-      [agreementState.active, P.any, P.any, true],
-      () => agreementState.suspended
-    )
+  match([stateByAttribute, suspendedByProducer, suspendedByConsumer])
+    .with([agreementState.active, true, P.any], () => agreementState.suspended)
+    .with([agreementState.active, P.any, true], () => agreementState.suspended)
```

Comment:
Is it correct that they return the same state in both cases? Maybe we can simplify this logic if that's the case

Reply:
Yes it's the right behavior.

Reply:
I changed a bit the code to improve the readability